### ISSUE TITLE
Healthy/unhealthy checks don't count consecutive successes/failures

### ISSUE THREAD

**Issue by mabeyj** [blindsidenetworks/scalelite#251]
Looking at the poller code, it only resets the healthy/unhealthy counters when a threshold is crossed:

https://github.com/blindsidenetworks/scalelite/blob/8b4830ced5a609028fed6270efd14e8bb11a8615/lib/tasks/poll.rake#L44-L47
https://github.com/blindsidenetworks/scalelite/blob/8b4830ced5a609028fed6270efd14e8bb11a8615/lib/tasks/poll.rake#L57-L61

Using the default `SERVER_UNHEALTHY_THRESHOLD=2`, let's say a server has an intermittent issue which causes the poller to increment the unhealthy counter only once, then the server starts responding successfully again. Scalelite doesn't take the server offline (as expected), but the counter is not reset and it remains at 1. The next time an intermittent issue happens, the counter in incremented again and it is now 2, so Scalelite marks the server offline which is premature.

In other words, it's counting failures over any length of time rather than consecutive failures.

Similarly, this would happen for the healthy check that marks a server online if `SERVER_HEALTHY_THRESHOLD` is changed from the default of 1.

Real world example: I have Scalelite/BBB servers that are scheduled to run certain hours every day. Scalelite is running just before BBB is booted which usually causes the poller to pick up one failure. Yesterday, it counted one failure and the server remained online. Today, it also counted one failure and the server was taken offline (and disabled due to https://github.com/blindsidenetworks/scalelite/issues/248).
